library/x509.c

```diff
@@ -775,17 +775,24 @@ int mbedtls_x509_dn_gets( char *buf, size_t size, const mbedtls_x509_name *dn )
             ret = mbedtls_snprintf( p, n, "\?\?=" );
         MBEDTLS_X509_SAFE_SNPRINTF;
 
-        for( i = 0; i < name->val.len; i++ )
+        for( i = 0, j = 0; i < name->val.len; i++, j++ )
         {
-            if( i >= sizeof( s ) - 1 )
+            if( j >= sizeof( s ) - 1 )
                 break;
 
             c = name->val.p[i];
+            // Requires escaping, RFC 1779
+            if( c == ',' )
```

Comment:
RFC 1779 lists a number of special characters - should we handle just comma, all of them, or a subset?

`<special> ::= "," | "=" | <CR> | "+" | "<" |  ">" | "#" | ";"`

Reply:
Currently `mbedtls_x509_string_to_names` only handles escaped commas, so this would need changes to handle the others. I think it would make sense to handle all of the special characters the same way to be consistent.